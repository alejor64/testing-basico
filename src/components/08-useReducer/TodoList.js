import React from 'react'
import TodoListoItem from './TodoListoItem'

const TodoList = ({todos, handleToggle, handleDelete}) => {
    return (
        <ul className="list-group list-group-flush">
           {
                todos.map((todo, i) => (
                    <TodoListoItem
                        key={todo.id}
                        i={i} 
                        todo={todo} 
                        handleToggle={handleToggle}
                        handleDelete={handleDelete}
                    />
                ))
            }
        </ul>
    )
}

export default TodoList
