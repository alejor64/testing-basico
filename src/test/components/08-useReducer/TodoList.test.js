import React from 'react'
import {shallow} from 'enzyme'
import '@testing-library/jest-dom'
import { demoTodo } from '../../fixtures/demoTodo'
import TodoList from '../../../components/08-useReducer/TodoList'

describe('Pruebas en el componente <TodoList />', () => {
    const handleDelete = jest.fn()
    const handleToggle = jest.fn()
    const wrapper = shallow(
        <TodoList 
            todos={demoTodo}
            handleToggle={handleToggle}
            handleDelete={handleDelete}
        />
    )

    test('Debe retornar el componente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('Debe de tener el componente <TodoListoItem />', () => {
        const TodoListoItem = wrapper.find('TodoListoItem')
        expect(TodoListoItem.length).toBe(demoTodo.length)
        expect(TodoListoItem.at(0).prop('handleToggle')).toEqual(expect.any(Function))
        expect(TodoListoItem.at(0).prop('handleDelete')).toEqual(expect.any(Function))
    })
    
})
