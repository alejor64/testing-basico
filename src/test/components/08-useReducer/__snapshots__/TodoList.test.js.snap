// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <TodoList /> Debe retornar el componente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListoItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListoItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    i={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender Mongo",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
